corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -204,17 +206,20 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
         },
 
         _isValid: function () {
+            if (this.model.get("error")) {
+                return false;
+            }
             if (!this.model.get('required')) {
                 return true;
             }
-
             var answer = this.getEncodedValue();
             return answer !== undefined && (answer === "" || answer.replace(/\s+/, "") !== "");
         },
 
         isValid: function () {
             var hasError = !this._isValid();
             if (hasError !== this.hasError) {
+                this.errorMessage = this.model.get("error");
```

Comment:
>[So does this mean that formplayer will always provide a message here, then?](https://github.com/dimagi/commcare-hq/pull/31879#discussion_r920304529)

@shubham1g5, noting this here since the call to `model.get("error")` has moved. Thinking about @esoergel's question, that was my interpretation. Is that correct?

Reply:
@Robert-Costello yes, there should always be a `error` message in case of a validation error.